pypeit/core/pca.py

```diff
@@ -255,21 +258,24 @@ def fit_pca_coefficients(coeff, order, ivar=None, weights=None, function='legend
     fit_coeff = [None]*npca
 
     # TODO: This fitting is fast. Maybe we should determine the best
-    # order for each PCA component, up to some maximum, by comparing
-    # reduction in chi-square vs added number of parameters?
+    #  order for each PCA component, up to some maximum, by comparing
+    #  reduction in chi-square vs added number of parameters?
 
     # Fit the coefficients of each PCA component so that they can be
     # interpolated to other coordinates.
 
     inmask = np.ones_like(coo, dtype=bool)
+    pypeitFits = []
     for i in range(npca):
-        coeff_used[:,i], fit_coeff[i] \
-                = utils.robust_polyfit_djs(coo, _coeff[:,i], _order[i], inmask=inmask,
+        #coeff_used[:,i], fit_coeff[i] \
+        pypeitFit = fitting.robust_fit(coo, _coeff[:,i], _order[i], in_gpm=inmask,
                                            invvar=None if _ivar is None else _ivar[:,i],
                                            weights=_weights[:,i], function=function,
                                            maxiter=maxiter, lower=lower, upper=upper,
                                            maxrej=maxrej, sticky=False, use_mad=_ivar is None,
                                            minx=minx, maxx=maxx)
+        coeff_used[:,i], fit_coeff[i] = pypeitFit.gpm, pypeitFit.fitc
+        pypeitFits.append(pypeitFit)
```

Comment:
I'm not actually sure but I see your point. For now I added the deepcopy to be safe. I believe the numpy array assignments are fine since those don't use pointers.